plugins/woocommerce/includes/abstracts/abstract-wc-product.php

```diff
@@ -2004,20 +2004,57 @@ public function get_formatted_name() {
 	 * Get min quantity which can be purchased at once.
 	 *
 	 * @since  3.0.0
-	 * @return int
+	 * @return int|float
 	 */
 	public function get_min_purchase_quantity() {
-		return 1;
+		/**
+		 * Filters the minimum quantity which can be purchased at once.
+		 *
+		 * @since 10.1.0
+		 *
+		 * @param int|float $quantity The minimum quantity.
+		 * @param WC_Product $this The product object.
+		 * @return int|float The minimum quantity.
+		 */
+		return wc_stock_amount( apply_filters( 'woocommerce_quantity_input_min', 1, $this ) );
```

Comment:
This filter was present in the qty input, but I wanted to consolidate the hooks rather than add new ones here. This means that `woocommerce_quantity_input_min`, `woocommerce_quantity_input_max`, `woocommerce_quantity_input_step` can be used to modify the quantity inputs on classic cart, product pages, and Store API all at once.